### PR TITLE
HParams: Update scalar card to read runs with respect for hparams filters

### PR DESCRIPTION
**Pull request by rileyajones**
## Motivation for features / changes
As part of the effort to bring hparams data to the time series dashboard we are expanding the hparams filtering functionality in the runs table to effect the runs being shown in rest of the dashboard

Now that #6366 is merged, the rest of the dashboard can now access the list of runs with respect for any hparam filters a user has applied.

## Screenshots of UI changes (or N/A)
No Filters
image

Several HParam Filters Applied
image

Regex and HParam Filter Applied
image

Only  a Regex Filter Applied
image

All Runs Selected
image

1 Run Deselected
image

## Detailed steps to verify changes work correctly (as executed by you)
1) Start tensorboard with a logdir which contains hparam data
2) Navigate to http://localhost:6006?enableHparamsInTimeSeries
3) Filter the runs table by regex and ensure scalar cards reflect this
4) Unselect some runs and ensure scalar cards reflect this
5) Filter the runs table by an hparam and ensure scalar cards reflect this.

### CODE REVIEW

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts [hunk 1]
```diff
@@ -482,6 +478,10 @@ describe('scalar card', () => {
         selectors.getCurrentRouteRunSelection,
         new Map([['run1', true]])
       );
+      store.overrideSelector(
```
Comment (JamesHollyer) [inline tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts:481]: I wish these could all go in to before each sections. However, I understand that the run is set here and it does make sense to do this after that.

**Review by JamesHollyer — APPROVED** [review]

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts [hunk 2]
```diff
@@ -57,6 +56,7 @@ import {
   getMetricsCardRangeSelectionEnabled,
   getRun,
   getRunColorMap,
+  getCurrentRouteRunSelection,
```
[the remark this answers is not in the document]
Comment (rileyajones) [reply tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts:59]: I removed it and reimported it and expected prettier to put it in the same place 🤷
Something definitely seems off with prettier lately

File: tensorboard/webapp/metrics/views/main_view/common_selectors.ts [hunk 3]
```diff
@@ -244,6 +246,18 @@ export const getFilteredRenderableRunsFromRoute = createSelector(
...
 );
 
+export const getFilteredRenderableRunsIdsFromRoute = createSelector(
+  getFilteredRenderableRunsFromRoute,
+  (filteredRenderableRuns) => {
+    return new Set(filteredRenderableRuns.map(({run: {id}}) => id));
+  }
+);
+
+export const factories = {
```
[the remark this answers is not in the document]
Comment (rileyajones) [reply tensorboard/webapp/metrics/views/main_view/common_selectors.ts:256]: In order for the tests to work properly they need to spyOn the same object as the is being used by the code they are testing. There are a few oddities when it comes to testing selector factories, I want to add some additional utilities to help with that in the future

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts (hunk 1 above)
Comment (rileyajones) [reply tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts:481]: I definitely thought about it, but as you suspected I wanted it to mirror the run selection override.

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts [hunk 4]
```diff
@@ -2434,6 +2439,7 @@ describe('scalar card', () => {
           end: null,
         });
         store.refreshState();
+        tick();
```
[the remark this answers is not in the document]
Comment (rileyajones) [reply tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts:2442]: I don't like this either, it does only add ~2ms to the test runtime, but it feels janky. It has something to do with the underlying selector needing to be recreated. I have an idea on how to solve this sort of issue, but I'd like to handle that in a separate PR

**Pull request merged**